demo/admin/src/products/ProductForm.tsx

```diff
@@ -331,6 +331,7 @@ export function ProductForm({ id }: FormProps) {
                             multiple
                             maxFileSize={1024 * 1024 * 4} // 4 MB
                             fullWidth
+                            layout="grid"
```

Comment:
I'm not sure about `layout=grid` being the switch between preview or preview-less mode. I'd prefer something more explicit like `previewImages=true`

Reply:
@jamesricky what do you think?

Reply:
Do you mean we might need to show an actual preview-image in the list-view in the future? 
I don't think that will be the case. At least all designs only show the preview in the grid view.

Reply:
> Do you mean we might need to show an actual preview-image in the list-view in the future?

Yes.

>  I don't think that will be the case. At least all designs only show the preview in the grid view.

It's not the first time that designs have changed considerably 😁 Maybe the next layout will be "imageCarousel" or whatever. We'd than need an additional showAsCarouselWithPreview

Reply:
Alternative idea: Switch to layout="grid" automatically if any of the files has a preview URL and layout hasn't been explicitly set to "list"?

Reply:
> Alternative idea: Switch to layout="grid" automatically if any of the files has a preview URL and layout hasn't been explicitly set to "list"?

The designers created this as two separate components actually: 
- A file upload, which includes a list of files that have been uploaded
- A grid of preview images

They never intended combining those in any way, so I think it should explicitly be set which variant to show. 

The reason we combined them (initially I think they were separate) was that they both needed to access the data in the form state and handle that data in the same way. 

Maybe it makes sense to separate them after all 🤔 

> Maybe the next layout will be "imageCarousel"

That is actually in the design and will probably come in the future but it's part of the grid variant, and opens when you click on one of the images in a sort of modal 😄

Reply:
So showAsGridWithPreview wouldn't solve the problem either? 😬 

Maybe we should change this in a follow-up PR?

Reply:
> Maybe we should change this in a follow-up PR?

Yes, maybe just leave it as is for now and we can discuss this in person another time

Reply:
@thomasdax98 would you be okay with this?

Reply:
That's fine by me